### ISSUE TITLE
Crash on newly added device

### ISSUE THREAD

**Issue by MeFisto94** [libratbag/piper#115]
Hey,
I've added ratbag support for my G703 by adding it as hid++ 2.0 to udev, so that it now shows up in piper, but when I try to click it's name, I get:

```
PYTHONPATH=$PYTHONPATH:/usr/lib/python3.5/site-packages piper
Device Logitech USB Receiver has no image or its path is invalid
Device Logitech Performance MX has no image or its path is invalid

(piper:20328): Gtk-WARNING **: Theme directory base/ of theme oxygen has no size field

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/gi/_propertyhelper.py", line 427, in obj_set_property
    prop.fset(self, value)
  File "/home/x/Dokumente/02_GIT/piper/piper/mouseperspective.py", line 87, in device
    self.stack.add_titled(ResolutionsPage(device), "resolutions", _("Resolutions"))
  File "/home/x/Dokumente/02_GIT/piper/piper/resolutionspage.py", line 66, in __init__
    self._init_ui()
  File "/home/x/Dokumente/02_GIT/piper/piper/resolutionspage.py", line 71, in _init_ui
    mousemap = MouseMap("#Buttons", self._device, spacing=20, border_width=20)
  File "/home/x/Dokumente/02_GIT/piper/piper/mousemap.py", line 111, in __init__
    raise ValueError("Device has no image or its path is invalid")
ValueError: Device has no image or its path is invalid
```

Fun Fact: I can click on the Performance MX without that bug, despite the warning that no image would be available. It seems to me that I only miss the images related to the mousebuttons?

If so, is there some documentation on what steps to undergo to add support for a new device piper-wise?

**Comment by whot** [comment]
See libratbag/data/gnome/README.md for the instructions on how to add a compatible svg, once that's in and installed, piper should just work.

What version of piper are you running though? git master?

**Comment by Hjdskes** [comment]
Can you elaborate (maybe with screenshots) what you mean with

> Fun Fact: I can click on the Performance MX without that bug, despite the warning that no image would be available.

Also please open a separate issue, or, better yet, a pull request, on the libratbag repository for adding support and a device SVG for the Performance MX.

**Comment by MeFisto94** [comment]
Oh I didn't know that the SVG is related to libratbag actually.
I didn't add the Performance MX, it was already there, however shows an empty page when clicked, which only shows "Profile" to it's left and that's it.

As for the G703, I basically copied the G403's entry (especially since they nearly look identical):
```
+# G703 over Wireless
+mouse:usb:v046dpc539:name:*:
+ RATBAG_DRIVER=hidpp20
+ RATBAG_HIDPP20_INDEX=1
+ RATBAG_SVG=logitech-g403.svg
```
So I am uncertain why it should crash, given that a G403 would work (I can't test that).
I confirmed that logitech-g403.svg is in the data/gnome folder, but not in data/default. I am using KDE, I don't know if that is relevant though?

On the other hand it seems the G703 reporting buttons not present in the G403's SVG?

Btw: After a reboot libratbag isn't discovering the devices anymore, so I might not get back to you that fast, as I try to fix that first (probably removing solaar destroyed something)

**Comment by Hjdskes** [comment]
> I didn't add the Performance MX, it was already there, however shows an empty page when clicked, which only shows "Profile" to it's left and that's it.

That's currently what happens when a device SVG cannot be found; I should fix this to display the error perspective.

> So I am uncertain why it should crash, given that a G403 would work (I can't test that).

It works, it's the one I'm testing Piper with.

> I confirmed that logitech-g403.svg is in the data/gnome folder, but not in data/default. I am using KDE, I don't know if that is relevant though?

It doesn't matter, and indeed the G403 SVG is missing from data/default but for Piper, this doesn't matter.

> On the other hand it seems the G703 reporting buttons not present in the G403's SVG?

This is a libratbag issue, please open an issue for this and adding a G703 entry on the libratbag repository.

I'm keeping this issue open as a reminder to open the error perspective when a device's SVG cannot be found.

**Comment by Hjdskes** [comment]
Fixed in #125, closing.
